jupyterlab_templates/extension.py

```diff
@@ -5,72 +5,56 @@
+
 import jupyter_core.paths
 import tornado.web
 
-from io import open
 from notebook.base.handlers import IPythonHandler
 from notebook.utils import url_path_join
 
+from pyarrow import fs
+
 
 class TemplatesLoader:
-    def __init__(self, template_dirs):
+    def __init__(self, template_dirs, log):
         self.template_dirs = template_dirs
+        self.log = log
 
     def get_templates(self):
-        templates = {}
+        templates = defaultdict(list)
         template_by_path = {}
 
-        for path in self.template_dirs:
-            # in order to produce correct filenames, abspath should point to the parent directory of path
-            abspath = os.path.abspath(os.path.join(os.path.realpath(path), os.pardir))
-            files = []
-            # get all files in subdirectories
-            for dirname, _, filenames in os.walk(path, followlinks=True):
-                if dirname == path:
-                    # Skip top level
-                    continue
-
-                for filename in fnmatch.filter(filenames, "*.ipynb"):
-                    if ".ipynb_checkpoints" not in dirname:
-                        files.append(
-                            (
-                                os.path.join(dirname, filename),
-                                dirname.replace(path, ""),
-                                filename,
-                            )
-                        )
-            # pull contents and push into templates list
-            for f, dirname, filename in files:
-                # skips over faild attempts to read content
-                try:
-                    with open(os.path.join(abspath, f), "r", encoding="utf8") as fp:
-                        content = fp.read()
-                except (FileNotFoundError, PermissionError):
-                    # Can't read file, skip
-                    continue
-
-                data = {
-                    "path": f,
-                    "name": os.path.join(dirname, filename),
-                    "dirname": dirname,
-                    "filename": filename,
-                    "content": content,
-                }
-
-                # remove leading slash for select
-                if dirname.strip(os.path.sep) not in templates:
-                    templates[dirname.strip(os.path.sep)] = []
-
-                # don't include content unless necessary
-                templates[dirname.strip(os.path.sep)].append({"name": data["name"]})
-
-                # full data
-                template_by_path[data["name"]] = data
+        for uri in self.template_dirs:
+            try:
+                client, path = fs.FileSystem.from_uri(uri)
+            except Exception as e:
+                self.log.error("Failed to load template directory %s. \n%s", uri, e)
+                continue
+
+            try:
+                for file in client.get_file_info(fs.FileSelector(path, recursive=True)):
+                    if file.extension == 'ipynb':
```

Comment:
```suggestion
                    if file.extension == "ipynb":
```
To follow the existing style of the repo